flax/struct.py

```diff
@@ -39,7 +39,24 @@
 import jax
 
 
-def dataclass(clz: type):
+
+# This decorator is interpreted by static analysis tools as a hint
+# that a decorator or metaclass causes dataclass-like behavior.
+_T = TypeVar("_T")
+def __dataclass_transform__(
```

Comment:
Should we really add dunders to this?

I don't know that from other libraries, maybe you found some guidance on when to use them?

Reply:
Yes, I've added a link to the proposal that explains this magic